packages/storybook/scripts/build.mjs

```diff
@@ -10,7 +10,9 @@ async function main() {
     console.log('Skipping storybook build for stable build');
     return;
   }
-  await exec(quote(['npx', 'build-storybook', '--quiet', '--loglevel', 'warn', ...process.argv.slice(2)]));
+  await exec(quote(['npx', 'build-storybook', '--quiet', '--loglevel', 'warn', ...process.argv.slice(2)]), {
+    NODE_OPTIONS: '--openssl-legacy-provider'
```

Comment:
can you add comments why we have the NODE_OPTIONS here (i.e. to support node 18 and should be removed if we update to storybook 7)